File: Tests/Gemini/QuixBugs/Gemini_1.0_Pro_Java/OS/java_programs/POWERSET.java
package java_programs;
import java.util.*;


public class POWERSET {
    public static ArrayList<ArrayList> powerset(ArrayList arr) {
        if (!arr.isEmpty()) {
            Object first = arr.get(0);
            arr.remove(0);
            ArrayList rest = arr;
            ArrayList<ArrayList> rest_subsets = powerset(rest);

            ArrayList<ArrayList> output = new ArrayList<ArrayList>(100);


            for (ArrayList subset : rest_subsets) { // Line changed here
                ArrayList to_add = new ArrayList(100); // Line changed here
                to_add.add(first);
                to_add.addAll(subset);
                output.add(to_add);
            }

            return output;
        } else {
            ArrayList empty_set = new ArrayList<ArrayList>();
            empty_set.add(new ArrayList());
            return empty_set;
        }
    }
}

File: Conversation_Outputs/Guided_Convergence_Code/Python/powerset.py
def powerset(arr):
    if arr:
        first, *rest = arr 
        rest_subsets = powerset(rest)
        # Fix: Prepend subsets without 'first' to maintain order
        return rest_subsets + [[first] + subset for subset in rest_subsets]  
    else:
        return [[]]